### ISSUE TITLE
Change Shadowed Fields on Existing Core Filter Sets

### ISSUE THREAD

**Issue by bryanculver** [nautobot/nautobot#2806]
Adapted from https://gist.github.com/jathanism/7c585954da4cb1a54997be31991cdaac#coreutilities --

## Core/Utilities

Utilities are being included with core since `nautobot.utilities.filters.BaseModelSet` is the bread and butter here and we may very well end up folding all of the primitives from Utilities into Core to streamline the namespace.

Items that have been moved to their own story:
### BaseFilterSet

- The model field to filter field mapping at `BaseFilterSet.FILTER_DEFAULTS` needs to also include the correct UI widget types in the `extra` for the filter mapping
    - #3582
    - #3583
- Filter mappings for ALL bespoke field types should be extended
    - For example `TaggableManager` should always be mapped to `TagFilter` to obviate the need to define `tag`/`tags` on every new filterset that requires tags. (but see also #3154)
    - #3584
- ~The base form used should be included by default on the [`Meta.form`](https://django-filter.readthedocs.io/en/stable/ref/filterset.html#form) option. This base form should be what is used as the base form for our UI including its own [model field to form field mappings](https://docs.djangoproject.com/en/4.0/topics/forms/modelforms/#overriding-the-default-fields). (See `field_classes` on that page).~
    - Most to be done with the JSON schema representation but we may need to somehow provide a way when the schema isn't good enough to specifically provide the react form that represents that model (either by convention in the React packaging or by configuration in the model). Let's start with the schema first.
    - Let's see how much `.form` is used anymore with the new UI
- #3585
- #3587
- #3588

--------

Review:

https://gist.github.com/jathanism/7c585954da4cb1a54997be31991cdaac#extras

for fields to be changed under "Incorrect Fields"

**NOTE:** Let's do this after #2721 ~and #2722~

**Comment by bryanculver** [comment]
Blocked by #2721 and #2722

**Comment by glennmatthews** [comment]
Since we decided not to collapse extras into core for the time being, should this be split into two issues?

**Comment by bryanculver** [comment]
Done because bullet points were either already done or opened as separate issue, many of which are enhancements and not breaking changes.

💪
